packages/ui/src/common/hooks/useRefetch.ts

```diff
@@ -1,24 +1,21 @@
-import { useContext, useEffect, useState } from 'react'
+import { useContext, useEffect } from 'react'
 
 import { RefetchContext, UseRefetch } from '../providers/refetch/context'
 
 type Action = { type: 'set'; payload: UseRefetch[0] } | { type: 'do'; payload?: boolean }
 
 export const useRefetch = (action: Action) => {
   const [refetch, setRefetch] = useContext(RefetchContext)
-  const [refetching, setRefetching] = useState<Promise<any> | undefined>()
 
   useEffect(() => {
     switch (action.type) {
       case 'set':
         return setRefetch(action.payload)
       case 'do':
         if (action.payload) {
-          setRefetching(refetch?.())
+          refetch?.()
```

Comment:
There was a problem with an infinite loop because of setting state inside useEffect. Probably it could be fixed quite easily, but it was also kind of misleading naming to me (refetch + refetching) so I left it in a simpler way for now